### PR TITLE
[#169] slip rate factors

### LINKED ISSUE #169: Ability to scale slip rate in TVZ

### PR DESCRIPTION
**Pull request by voj**
closes #169

### CODE REVIEW

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_InversionFaultSystemRuptSet.java [hunk 1]
```diff
@@ -79,11 +74,13 @@ protected static FaultSystemRupSet prepCrustalRupSet(FaultSystemRupSet rupSet, N
 			scaling.setRecalc(true);
 		}
 
-		if (scaling != null && scaling.getReCalc()) {
+		if (scaling.getReCalc()) {
 			rupSet = recalcMags(rupSet, scaling);
 		}
 
 		if (magBounds != null && magBounds.getMaxMagType() == NZSHM22_MagBounds.MaxMagType.FILTER_RUPSET) {
+			rupSet.addModule(faultPolyMgr(rupSet, branch));
```
Comment (chrisbc) [inline src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_InversionFaultSystemRuptSet.java:82]: @voj @chrisdicaprio agreed to ditch this feature

Comment (chrisbc) [reply src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_InversionFaultSystemRuptSet.java:82]: :)

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_InversionFaultSystemRuptSet.java [hunk 2]
```diff
@@ -117,16 +114,33 @@ public static NZSHM22_InversionFaultSystemRuptSet fromExistingCrustalSet(FaultSy
...
+		}
+
+		NZSHM22_TvzSections tvzSections = rupSet.getModule(NZSHM22_TvzSections.class);
+		SectSlipRates origSlips = rupSet.getModule(SectSlipRates.class);
+		double[] slipRates = origSlips.getSlipRates();
+
+		if (factors.getTvzFactor() >= 0) {
+			for (int i = 0; i < slipRates.length; i++) {
+				if (tvzSections.isInRegion(i)) {
+					slipRates[i] *= factors.getSansFactor();
```
Comment (chrisbc) [inline src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_InversionFaultSystemRuptSet.java:130]: TVZ!!!

File: src/test/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_InversionFaultSystemRuptSetTest.java [hunk 3]
```diff
@@ -38,4 +40,27 @@ public void recalcMagsTest() throws DocumentException, URISyntaxException, IOExc
...
+        branch.setValue(factors);
+
+        NZSHM22_InversionFaultSystemRuptSet.applySlipRateFactor(rupSet, branch);
+        double[] actual = rupSet.getModule(SectSlipRates.class).getSlipRates().clone();
+
+        for(int i =0; i < original.length; i++){
+            original[i]*= 0.3;
+        }
+
+        // TODO use a rupture set that also has TVZ sections
```
Comment (chrisbc) [inline src/test/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_InversionFaultSystemRuptSetTest.java:62]: Let's add that now @voj

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_InversionFaultSystemRuptSet.java (hunk 2 above)
Comment (voj) [reply src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_InversionFaultSystemRuptSet.java:130]: whoops!

**Comment by voj** [comment]
merged

**Pull request closed without merging**